lib/elastic_record/index.rb

```diff
@@ -25,30 +23,30 @@ module ElasticRecord
   #   Returns if the index exists
   # [get_mapping]
   #   Returns the mapping currently stored by elastic search.
-  # [put_mapping]
+  # [update_mapping]
   #   Update elastic search's mapping
   class Index
     include Documents
     include Manage
     include Mapping, Settings
-    include Percolator
     include Analyze
     include Deferred
 
-    attr_accessor :model
+    attr_accessor :doctypes
 
     attr_accessor :disabled
-    attr_accessor :has_percolator
+    attr_accessor :model
     attr_accessor :partial_updates
 
-    def initialize(model)
-      @model = model
+    def initialize(models)
+      models = Array.wrap(models)
+      @model = models.first
+      @doctypes = models.map(&:doctype)
       @disabled = false
     end
```

Comment:
The goal was to model query documents as regular documents.  This has the advantage of eliminating all of the special percolator-specific code.  Indexes with percolators now just have two doctypes, and the necessary analysis settings and mappings are built automatically.

From elasticsearch's perspective, multiple doctype support should actually [be expected!](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping.html#mapping-type)  We are restricting ElasticSearch by omitting it.

But, yes, from the `ElasticRecord::Model`'s perspective, doctype specification is redundant.